backend/src/Designer/TypedHttpClients/DelegatingHandlers/CachingDelegatingHandler.cs

```diff
@@ -46,11 +47,29 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                 Data = await response.Content.ReadAsByteArrayAsync(cancellationToken),
                 StatusCode = response.StatusCode
             };
-            _memoryCache.Set(cacheKey, newCacheEntry, new MemoryCacheEntryOptions().SetAbsoluteExpiration(TimeSpan.FromSeconds(_cacheExpiryInSeconds)));
+
+            MemoryCacheEntryOptions cacheEntryOptions = GenerateMemoryCacheEntryOptions();
+
+            _memoryCache.Set(cacheKey, newCacheEntry, cacheEntryOptions);
             response.Dispose();
             return GetCachedResponseMessage(newCacheEntry);
         }
 
+        private MemoryCacheEntryOptions GenerateMemoryCacheEntryOptions()
+        {
+            var cancellationTokenSource = new CancellationTokenSource(
+                TimeSpan.FromSeconds(_cacheExpiryInSeconds));
+
+            var cacheEntryOptions = new MemoryCacheEntryOptions()
+                .AddExpirationToken(
+                    new CancellationChangeToken(cancellationTokenSource.Token))
+                .RegisterPostEvictionCallback((key, value, reason, state) =>
```

Comment:
Could we use _ for the parameters; key, value and reason, since they are not relevant in this context? 😊

Reply:
Yes. I keept them since it's easier to change method if we need to tweak something for eviction.